### PR TITLE
Topnav balance [WAIV-1354]

### PR DESCRIPTION
**Pull request by konstantin1390**
Fixes #

### Changes

Summary of changes you made.

* Change 1
* Change 2

### Test plan

Explain how to test changes you made.

* [ ] Step 1
* [ ] Step 2

### Demo (optional)

Sreenshots (before & after) or video that show result of your work.

### CODE REVIEW

File: src/client/components/Navigation/Topnav.js [hunk 1]
```diff
@@ -50,6 +52,8 @@ import { getTopPosts } from '../../../waivioApi/ApiClient';
 import ModalSignUp from '../Authorization/ModalSignUp/ModalSignUp';
 import ModalSignIn from '../Authorization/ModalSignIn/ModalSignIn';
 import './Topnav.less';
+import { getHoldingsWithLogo } from '../../user/usersHelper';
+import CurrencyItem from '../../wallet/CurrencyItem/CurrencyItem';
```
Comment (milklim) [inline src/client/components/Navigation/Topnav.js]: import styles at the last line

File: src/client/components/Navigation/Topnav.js [hunk 2]
```diff
@@ -980,9 +1074,8 @@ class Topnav extends React.Component {
           </div>
           <div className="Topnav__right-bottom">
             {this.content()}
-
-            {isAuthenticated && (
-              <div className="Topnav__broker">
+            <div className={classNames('Topnav__broker', { 'justify-end': !isAuthenticated })}>
+              {!isAuthenticated ? (
```
Comment (milklim) [inline src/client/components/Navigation/Topnav.js]: display currencies block only if platform !== 'widgets'

File: src/client/user/UserWallet.js [hunk 3]
```diff
@@ -138,14 +138,13 @@ class Wallet extends Component {
   }
 
   getBeaxyBalanceWithLogo = () => {
-    const { currenciesDescriptions, beaxyBalance } = this.props;
```
Comment (konstantin1390) [inline src/client/user/UserWallet.js]: rename

File: src/client/locales/default.json [hunk 4]
```diff
@@ -665,6 +665,7 @@
   "headerAuthorized.freeBalance": "Free balance",
   "headerAuthorized.openDeals": "Open deals",
   "headerAuthorized.closedDeals": "Closed deals",
+  "headerAuthorized.allBalances": "All balances",
```
Comment (konstantin1390) [inline src/client/locales/default.json]: remove

File: src/client/components/Navigation/Topnav.js (hunk 1 above)
Comment (konstantin1390) [reply src/client/components/Navigation/Topnav.js]: ++

File: src/client/locales/default.json (hunk 4 above)
Comment (konstantin1390) [reply src/client/locales/default.json]: ++

File: src/client/user/UserWallet.js (hunk 3 above)
Comment (konstantin1390) [reply src/client/user/UserWallet.js]: ++

File: src/client/components/Navigation/Topnav.js [hunk 5]
```diff
@@ -856,6 +867,7 @@ class Topnav extends React.Component {
       openChat,
       messagesCount,
       platformName,
+      beaxyBalance,
```
Comment (milklim) [inline src/client/components/Navigation/Topnav.js]: remove as unused

File: src/client/components/Navigation/Topnav.js [hunk 6]
```diff
@@ -158,6 +165,10 @@ class Topnav extends React.Component {
       weeklyChosenPost: '',
       scrolling: false,
       visible: false,
+      initFirstCurrency: {},
+      initSecondCurrency: {},
+      firstCurrency: {},
+      secondCurrency: {},
```
Comment (milklim) [inline src/client/components/Navigation/Topnav.js]: remove as unused

File: src/client/components/Navigation/Topnav.js [hunk 7]
```diff
@@ -120,6 +123,8 @@ class Topnav extends React.Component {
     openChat: PropTypes.func.isRequired,
     messagesCount: PropTypes.number,
     isGuest: PropTypes.bool,
+    beaxyBalance: PropTypes.arrayOf(PropTypes.shape()),
+    getUserStatistics: PropTypes.func,
```
Comment (milklim) [inline src/client/components/Navigation/Topnav.js]: remove as unused

File: src/client/components/Navigation/Topnav.js [hunk 8]
```diff
@@ -78,6 +80,7 @@ import './Topnav.less';
     searchObjectTypesAutoCompete,
     resetSearchAutoCompete,
     toggleModal,
+    getUserStatistics,
```
Comment (milklim) [inline src/client/components/Navigation/Topnav.js]: unused - remove

File: src/client/components/Navigation/Topnav.js [hunk 9]
```diff
@@ -13,6 +13,7 @@ import {
   searchObjectTypesAutoCompete,
   searchUsersAutoCompete,
 } from '../../search/searchActions';
+import { getUserStatistics } from '../../../investarena/redux/actions/platformActions';
```
Comment (milklim) [inline src/client/components/Navigation/Topnav.js]: unused - remove

File: src/client/components/Navigation/Topnav.js [hunk 10]
```diff
@@ -980,22 +992,24 @@ class Topnav extends React.Component {
...
             {this.content()}
-
-            {isAuthenticated && (
-              <div className="Topnav__broker">
+            <div
+              className={classNames('Topnav__broker', {
+                'justify-end': !isAuthenticated || platformName === 'widgets',
+              })}
+            >
+              {!isAuthenticated || platformName === 'widgets' ? (
```
Comment (milklim) [inline src/client/components/Navigation/Topnav.js]: show broker button only for authenticated user

File: .gitignore [hunk 11]
```diff
@@ -40,3 +40,4 @@ node_modules
 .envrc
 
 build/
+/yarn.lock
```
Comment (milklim) [inline .gitignore]: do not commit .lock file

File: src/client/components/Navigation/BrokerBalance/BrokerBalance.js [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+import CurrencyItem from '../../../wallet/CurrencyItem/CurrencyItem';
+import { getUserStatistics } from '../../../../investarena/redux/actions/platformActions';
+import Loading from '../../Icon/Loading';
+import './BrokerBalance.less';
+
+const BrokerBalance = ({ beaxyBalance, platformName, getStatistics }) => {
+  const [initFirstCurrency, setInitFirstCurrency] = useState({});
+  const [initSecondCurrency, setInitSecondCurrency] = useState({});
+  const storageFirstItem = localStorage.getItem('firstCurrency');
+  const storageSecondCurrency = localStorage.getItem('secondCurrency');
```
Comment (milklim) [inline src/client/components/Navigation/BrokerBalance/BrokerBalance.js]: use 'store' lib to deal with local storage (https://github.com/Waiviogit/waivio/blob/dev-beaxy/src/client/helpers/localStorageHelpers.js)

File: src/client/components/Navigation/BrokerBalance/BrokerBalance.js [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+  }, [beaxyBalance]);
+
+  const setFirstCurrency = item => {
+    localStorage.setItem('firstCurrency', JSON.stringify(item));
+    setInitFirstCurrency(item);
+  };
+
+  const setSecondCurrency = item => {
+    localStorage.setItem('secondCurrency', JSON.stringify(item));
+    setInitSecondCurrency(item);
```
Comment (milklim) [inline src/client/components/Navigation/BrokerBalance/BrokerBalance.js]: use store.js to deal with local storage

File: src/client/components/Navigation/Topnav.js (hunk 5 above)
Comment (konstantin1390) [reply src/client/components/Navigation/Topnav.js]: ++

File: src/client/components/Navigation/Topnav.js (hunk 6 above)
Comment (konstantin1390) [reply src/client/components/Navigation/Topnav.js]: ++

File: src/client/components/Navigation/Topnav.js (hunk 7 above)
Comment (konstantin1390) [reply src/client/components/Navigation/Topnav.js]: ++

File: src/client/components/Navigation/Topnav.js (hunk 8 above)
Comment (konstantin1390) [reply src/client/components/Navigation/Topnav.js]: +

File: src/client/components/Navigation/Topnav.js (hunk 9 above)
Comment (konstantin1390) [reply src/client/components/Navigation/Topnav.js]: +

File: src/client/components/Navigation/Topnav.js (hunk 10 above)
Comment (konstantin1390) [reply src/client/components/Navigation/Topnav.js]: +++

**Pull request merged**